### PR TITLE
`vcd_vapp_vm` - do not check for None network records in vApp when used with `dhcp_wait_seconds`

### PR DESCRIPTION
**Pull request by Didainius**
vCD 10.1 does not return `None` network reference anymore for endpoint:
`https://XXX/api/vApp/vapp-9191a728-09de-474e-a022-5fb3027ae10b/networkConfigSection/`
This PR adds a workaround skip reliance on this record.

This is what versions prior to 10.1 returned for None networks:
```
    <NetworkConfig networkName="none">
        <Description>This is a special place-holder used for disconnected network interfaces.</Description>
        <Configuration>
            <IpScopes>
                <IpScope>
                    <IsInherited>false</IsInherited>
                    <Gateway>196.254.254.254</Gateway>
                    <Netmask>255.255.0.0</Netmask>
                    <Dns1>196.254.254.254</Dns1>
                </IpScope>
            </IpScopes>
            <FenceMode>isolated</FenceMode>
        </Configuration>
        <IsDeployed>false</IsDeployed>
    </NetworkConfig>
</NetworkConfigSection>
```

### CODE REVIEW

File: vcd/resource_vcd_vapp_vm.go [hunk 1]
```diff
@@ -1723,7 +1723,10 @@ func readNetworks(d *schema.ResourceData, vm govcd.VM, vapp govcd.VApp) ([]map[s
 
 		var ok bool
 		if singleNIC["type"], ok = vAppNetworkTypes[vmNet.Network]; !ok {
-			return []map[string]interface{}{}, fmt.Errorf("unable to determine vApp network type for: %s", vmNet.Network)
+			// Prior vCD 10.1  used to return a placeholder for none networks. It allowed to
+			// identify NIC type for types.NoneNetwork. This was removed in 10.1 therefore we when
```
Comment (lvirbalas) [inline vcd/resource_vcd_vapp_vm.go:1727]:
```suggestion
			// identify NIC type for types.NoneNetwork. This was removed in 10.1 therefore when
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,8 @@ IMPROVEMENTS:
 
 * `resource/vcd_network_routed`, `resource/vcd_network_direct`, and `resource/vcd_network_isolated` now support in place updates. [GH-465]
 * `vcd_vapp_network`, `vcd_vapp_org_network` has now missing import documentation [GH-481] 
+* `resource/vcd_vapp_vm` does not search/validate None network definition in vApp when
+  `dhcp_wait_seconds` parameter is used [GH-485]
```
Comment (lvirbalas) [inline CHANGELOG.md:8]: Maybe we can rephrase this to be clearer for the end user? Some thoughts:
- For a user it's not clear what "None network" is
- I think the message should also state that this is to support 10.1 behavior

File: vcd/resource_vcd_vapp_vm.go (hunk 1 above)
Comment (Didainius) [reply vcd/resource_vcd_vapp_vm.go]: Fixed.

File: CHANGELOG.md (hunk 2 above)
Comment (Didainius) [reply CHANGELOG.md]: Rephrased - do have a look if it is clearer.
